PagingWithNetworkSample/app/src/main/java/com/android/example/paging/pagingwithnetwork/reddit/repository/inMemory/byItem/SubRedditDataSourceFactory.kt

```diff
@@ -30,11 +31,16 @@ import java.util.concurrent.Executor
 class SubRedditDataSourceFactory(
         private val redditApi: RedditApi,
         private val subredditName: String,
-        private val retryExecutor: Executor) : DataSource.Factory<String, RedditPost>() {
-    val sourceLiveData = MutableLiveData<ItemKeyedSubredditDataSource>()
+        private val retryExecutor: Executor) : DataSource.Factory<String, RedditPost> {
+
+    private val _sourceLiveData = MutableLiveData<ItemKeyedSubredditDataSource>()
```

Comment:
same as above -  can you just override the inferred type?

Reply:
as above, the type of sourceLiveData would be LiveData, because that's how we would declare it.